atd-vze/src/views/Crashes/CrashChange.js

```diff
@@ -0,0 +1,1238 @@
+import React, { useState, useEffect } from "react";
+import { Link, Redirect } from "react-router-dom";
+import { gql } from "apollo-boost";
+import { withApollo } from "react-apollo";
+import { useQuery, useMutation } from "@apollo/react-hooks";
+import { useAuth0 } from "../../auth/authContext";
+import { MiniDiff } from "../../Components/MiniDiff";
+import axios from "axios";
+
+import "./crash.scss";
+
+import {
+  Card,
+  CardBody,
+  CardHeader,
+  Col,
+  Row,
+  Button,
+  Alert,
+  Badge,
+  Modal,
+  ModalHeader,
+  ModalBody,
+  ModalFooter,
+  Spinner,
+} from "reactstrap";
+
+import { AppSwitch } from "@coreui/react";
+import {
+  GET_CRASH_CHANGE,
+  GET_CRASH_SECONDARY_RECORDS,
+  CRASH_MUTATION_DISCARD,
+  RECORD_MUTATION_UPDATE,
+  RECORD_DELETE_CHANGE_RECORDS,
+  UPSERT_MUTATION_DUMMY,
+} from "../../queries/crashes_changes";
+import {
+  importantCrashFields,
+  crashFieldDescription,
+  piiFields,
+  notNullValues,
+} from "./crashFieldDescriptions";
+import { crashChangeQuotedFields } from "./crashChangeQuotedFields";
+import { redirectUrl } from "../../index";
+
+function CrashChange(props) {
+  const crashId = props.match.params.id;
+  const [redirect, setRedirect] = useState(false);
+  const [selectedFields, setSelectedFields] = useState([]);
+  const [recordData, setRecordData] = useState({});
+  const [recordSecondaryData, setRecordSecondaryData] = useState({});
+  // Arrays of column names
+  const [importantFieldList, setImportantFieldList] = useState([]);
+  const [differentFieldsList, setDifferentFieldsList] = useState([]);
+  const [selectableList, setSelectableList] = useState([]);
+  // Arrays of Components
+  const [importantFields, setImportantFields] = useState([]);
+  const [differentFields, setDifferentFields] = useState([]);
+  const [showFieldsDiffOnly, setShowFieldsDiffOnly] = useState(true);
+  // Modals
+  const [approveAllChanges, setApproveAllChanges] = useState(false);
+  const [discardAllChanges, setDiscardAllChanges] = useState(false);
+  const [clearAllSelections, setClearAllSelections] = useState(false);
+  const [savingChanges, setSavingChanges] = useState(false);
+  const [errorDialog, setErrorDialog] = useState(false);
+  const [saveStatus, setSaveStatus] = useState("");
+  const [errorMessage, setErrorMessage] = useState("");
+  // CR3 Availbable
+  const [cr3available, setCR3Available] = useState(false);
+  const [upsertRecordQuery, setUpsertRecordQuery] = useState(
+    UPSERT_MUTATION_DUMMY
```

Comment:
Do we want this Dummy mutation? Maybe I'm misunderstanding the purpose.

Reply:
It's in line #85: 

```
const [upsertRecordUpdates] = useMutation(
    gql`
      ${upsertRecordQuery}
    `
  );
```

From what I recall, I was having issues passing an empty string or a null value, because useMutation was checking syntax and making sure the query was ready to run, but at this point I don't have a GraphQL query even generated from my templates. So I created a dummy that would serve that purpose: keep useMutation at peace while I generate a query later on.

Reply:
ok oh, i see. ty!